Use SymfonyStyle for command output

OskarStark opened the pull request:
I started to implement `SymfonyStyle` to the commands, I will follow up with the other commands if you like this.

Here are some screenshots:
![screenshot 2017-08-29 15 18 23](https://user-images.githubusercontent.com/995707/29823920-5dcbfd10-8cd0-11e7-8d48-5479bfce0915.png)
![screenshot 2017-08-29 15 18 44](https://user-images.githubusercontent.com/995707/29823924-5dd16dae-8cd0-11e7-8284-ede4c1bd4aa2.png)
![screenshot 2017-08-29 15 20 44](https://user-images.githubusercontent.com/995707/29823921-5dccb8fe-8cd0-11e7-8805-c725f44647e8.png)
![screenshot 2017-08-29 15 32 25](https://user-images.githubusercontent.com/995707/29823923-5dd1292a-8cd0-11e7-91e8-7ce12b5d84a9.png)
![screenshot 2017-08-29 15 37 26](https://user-images.githubusercontent.com/995707/29823922-5dd04dca-8cd0-11e7-8bf9-b0fae02849ea.png)
![screenshot 2017-08-31 07 38 17](https://user-images.githubusercontent.com/995707/29910006-9235729e-8e28-11e7-8af7-b95bc9260a06.png)
![screenshot 2017-08-31 07 38 41](https://user-images.githubusercontent.com/995707/29910005-9235539a-8e28-11e7-8c43-d04944b769b2.png)
![screenshot 2017-08-31 07 41 12](https://user-images.githubusercontent.com/995707/29910012-92568a06-8e28-11e7-9893-82f13f2068dd.png)
![screenshot 2017-08-31 07 41 31](https://user-images.githubusercontent.com/995707/29910008-9250d8ea-8e28-11e7-8df4-739ce62fb26a.png)
![screenshot 2017-08-31 07 47 01](https://user-images.githubusercontent.com/995707/29910010-9251ab80-8e28-11e7-88d2-0842d781603f.png)
![screenshot 2017-08-31 07 51 58](https://user-images.githubusercontent.com/995707/29910007-924e7f8c-8e28-11e7-9926-9bd8da034e73.png)
![screenshot 2017-08-31 07 53 45](https://user-images.githubusercontent.com/995707/29910011-92526336-8e28-11e7-9ddb-c4d9616eedfa.png)
![screenshot 2017-08-31 08 20 17](https://user-images.githubusercontent.com/995707/29910009-9251caac-8e28-11e7-9760-40be2476064e.png)
![screenshot 2017-08-31 08 30 37](https://user-images.githubusercontent.com/995707/29910013-926900e6-8e28-11e7-9fd8-01b6e6d7d2dc.png)
![screenshot 2017-08-31 09 05 06](https://user-images.githubusercontent.com/995707/29912604-87e0684e-8e32-11e7-8ba1-4300ca3a0dfd.png)
![screenshot 2017-08-31 09 07 37](https://user-images.githubusercontent.com/995707/29912605-87e15740-8e32-11e7-82cb-21a9d18fffe0.png)

Is the branch the correct one or shall I switch?

Cheers, Oskar

OskarStark commented:
fyi @javiereguiluz ;-)

javiereguiluz commented on lib/Doctrine/ORM/Tools/Console/Command/SchemaTool/UpdateCommand.php:
```diff
@@ -124,35 +132,48 @@ protected function executeSchemaCommand(InputInterface $input, OutputInterface $
-        $output->writeln('           Use the incremental update to detect changes during development and use');
-        $output->writeln('           the SQL DDL provided to manually update your database in production.');
-        $output->writeln('');
+        $this->io->caution(<<<EOT
```
In case you prefer it, most methods accept both strings and arrays of strings, so this code could be:

```php
$this->io->caution([
    'This operation should not be executed in a production environment!',
    '',
    'Use the incremental update to detect changes during development and use',
    'the SQL DDL provided to manually update your database in production.',
]);
```

OskarStark replied:
I saw this afterwards 👍 

I can rework this if the guys like the PR and prefer the array syntax.

Thank you for your feedback!

lcobucci replied:
Array syntax looks better in my opinion

OskarStark replied:
done

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/SchemaTool/UpdateCommand.php:
```diff
@@ -39,6 +40,11 @@
     /**
+     * @var SymfonyStyle
+     */
+    protected $io;
```
Maybe `ui` or `userInterface` is a better name?

lcobucci replied:
Also, wouldn't it be better to have this property in the `AbstractCommand`

OskarStark replied:
I can change this ofc, but I use `$io` everywhere and it was mentioned in the official blog post, too. `$ui` is a bit missleading IMO

@javiereguiluz what do you think, how do you use it?

Official blog post: https://symfony.com/blog/new-in-symfony-2-8-console-style-guide

lcobucci replied:
I just suggested that because IMO we're actually interacting with the CLI user interface 😄

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/SchemaTool/UpdateCommand.php:
```diff
@@ -109,13 +115,15 @@ protected function configure()
      */
     protected function executeSchemaCommand(InputInterface $input, OutputInterface $output, SchemaTool $schemaTool, array $metadatas)
     {
+        $this->io = new SymfonyStyle($input, $output);
```
This could also be part of the `AbstractCommand#execute()`

OskarStark replied:
I moved it

lcobucci reviewed (changes requested):
@OskarStark some small suggestions here, thanks for your contribution, the command's output is indeed looking better.

In terms of branching, this is correct 😉.

lcobucci commented:
@OskarStark just a remark, please make sure you send test updates too.

OskarStark commented:
![screenshot 2017-08-31 07 38 17](https://user-images.githubusercontent.com/995707/29910006-9235729e-8e28-11e7-8af7-b95bc9260a06.png)
![screenshot 2017-08-31 07 38 41](https://user-images.githubusercontent.com/995707/29910005-9235539a-8e28-11e7-8c43-d04944b769b2.png)
![screenshot 2017-08-31 07 41 12](https://user-images.githubusercontent.com/995707/29910012-92568a06-8e28-11e7-9893-82f13f2068dd.png)
![screenshot 2017-08-31 07 41 31](https://user-images.githubusercontent.com/995707/29910008-9250d8ea-8e28-11e7-8df4-739ce62fb26a.png)
![screenshot 2017-08-31 07 47 01](https://user-images.githubusercontent.com/995707/29910010-9251ab80-8e28-11e7-88d2-0842d781603f.png)
![screenshot 2017-08-31 07 51 58](https://user-images.githubusercontent.com/995707/29910007-924e7f8c-8e28-11e7-9926-9bd8da034e73.png)
![screenshot 2017-08-31 07 53 45](https://user-images.githubusercontent.com/995707/29910011-92526336-8e28-11e7-9ddb-c4d9616eedfa.png)
![screenshot 2017-08-31 08 20 17](https://user-images.githubusercontent.com/995707/29910009-9251caac-8e28-11e7-9760-40be2476064e.png)
![screenshot 2017-08-31 08 30 37](https://user-images.githubusercontent.com/995707/29910013-926900e6-8e28-11e7-9fd8-01b6e6d7d2dc.png)

OskarStark commented:
Ok, lets check the build 😄

Ocramius commented on lib/Doctrine/ORM/Tools/Console/Command/ClearCache/CollectionRegionCommand.php:
```diff
@@ -36,21 +37,24 @@
     /**
+     * @var SymfonyStyle
+     */
+    protected $ui;
```
`private`

Ocramius replied:
From what I see, `ui` is not needed to be retained in internals: please convert it to a local variable wherever possible. If multiple `private` methods rely on it, please pass it around as a call parameter.

OskarStark replied:
done

Ocramius commented on lib/Doctrine/ORM/Tools/Console/Command/ClearCache/CollectionRegionCommand.php:
```diff
@@ -74,32 +78,34 @@ protected function configure()
         $cache       = $em->getCache();
 
-        if ( ! $cache instanceof Cache) {
+        if (!$cache instanceof Cache) {
```
Please revert these spacing changes

OskarStark replied:
done

Ocramius commented on lib/Doctrine/ORM/Tools/Console/Command/ClearCache/CollectionRegionCommand.php:
```diff
@@ -36,21 +37,24 @@
 class CollectionRegionCommand extends Command
 {
     /**
+     * @var SymfonyStyle
```
`SymfonyStyle|null`

Ocramius commented on lib/Doctrine/ORM/Tools/Console/Command/ClearCache/EntityRegionCommand.php:
```diff
@@ -36,20 +37,23 @@
 class EntityRegionCommand extends Command
 {
     /**
+     * @var SymfonyStyle
```
`SymfonyStyle|null`

Ocramius commented on lib/Doctrine/ORM/Tools/Console/Command/ClearCache/MetadataCommand.php:
```diff
@@ -39,23 +40,27 @@
 class MetadataCommand extends Command
 {
     /**
+     * @var SymfonyStyle
```
`SymfonyStyle|null`

OskarStark commented:
Hmm, this is a bit strange.
on my local machine, all tests are running, if my shell is big, if i minimize it it delivers the same error results like travis:
```
5) Doctrine\Tests\ORM\Tools\Console\Command\ClearCacheEntityRegionCommandTest::testClearCacheEntryName
Failed asserting that '\n
 // Clearing second-level cache entry for entity                                \n
 // "Doctrine\Tests\Models\Cache\Country" identified by                         \n
 // "1"                                                                         \n
\n
' contains " // Clearing second-level cache entry for entity "Doctrine\Tests\Models\Cache\Country" identified by".
```

The `SymfonyStyle` add line breaks if needed.

any idea @Ocramius @javiereguiluz ?

OskarStark commented:
I found this: https://stackoverflow.com/questions/39190610/how-to-test-console-command-output-when-terminal-widths-differ/40347595#40347595

`putenv('COLUMNS=80');` or 120?

WDYT?

OskarStark commented:
lets check if https://github.com/doctrine/doctrine2/pull/6655/commits/142055cf68bb2133680f68b45365fdac00ccab6a makes travis happy 🎉

OskarStark commented:
oskar.stark:~/dev/misc/doctrine2 (symfony-style)$ env COLUMNS=10 php -d memory_limit=-1 vendor/bin/phpunit --group=test
PHPUnit 6.4-ga1b5663a4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.1.8
Configuration: /Users/oskar.stark/dev/misc/doctrine2/phpunit.xml.dist

.F                                                                  2 / 2 (100%)

Time: 484 ms, Memory: 40.00MB

There was 1 failure:

1) Doctrine\Tests\ORM\Tools\Console\Command\ClearCacheCollectionRegionCommandTest::testClearByOwnerEntityClassName
Failed asserting that '\n
 // Cleari\n
 // ng    \n
 // second\n
 // -level\n
 // cache \n
 // for   \n
 // collec\n
 // tion  \n
 //       \n
 // "Doctr\n
 // ine\Te\n
 // sts\Mo\n
 // dels\C\n
 // ache\S\n
 // tate#c\n
 // ities"\n
 // </info\n
 // >     \n
\n
' contains " // Clearing second-level cache for collection "Doctrine\Tests\Models\Cache\State#cities"".

/Users/oskar.stark/dev/misc/doctrine2/tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheCollectionRegionCommandTest.php:79

FAILURES!
Tests: 2, Assertions: 6, Failures: 1.
oskar.stark:~/dev/misc/doctrine2 (symfony-style)$ env COLUMNS=120 php -d memory_limit=-1 vendor/bin/phpunit --group=test
PHPUnit 6.4-ga1b5663a4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.1.8
Configuration: /Users/oskar.stark/dev/misc/doctrine2/phpunit.xml.dist

..                                                                  2 / 2 (100%)

Time: 474 ms, Memory: 40.00MB

OK (2 tests, 6 assertions)
oskar.stark:~/dev/misc/doctrine2 (symfony-style)$

OskarStark commented:
Looks like 2 builds are still failing.... I don't know why.

Scrutinizer tests pass, after setting the COLUMNS env in `.scrutinizer.yml`

OskarStark commented:
https://github.com/doctrine/doctrine2/pull/6655/files#diff-354f30a63fb0907d4ad57269548329e3R12 fixes most of the travis builds, but `7.1` and `7.2` builds are still failing and I don't know why 😕 ❗️

OskarStark commented:
I rebased my branch, but don't know if travis is working now

OskarStark commented:
The `7.1` and `7.2` are failing if there is a custom `MYSQL_VERSION` set.
Looks like the `COLUMNS` env var is ignored, do you spawn any sub commands or sth like this, where the `COLUMNS` env var is not used anymore?

At the start of the build, the `COLUMNS` env var is exported
![screenshot 2017-11-09 08 28 00](https://user-images.githubusercontent.com/995707/32593401-0fc5b58a-c528-11e7-81c6-fbce0b5208ca.png)

OskarStark commented:
any news?

OskarStark commented:
Not a single comment/feedback since the end of august, that makes me sad.... 😞 

If you don't like it I will close the PR, otherwise I need some help how to fix it

Ocramius reviewed (approved):
Didn't get to do any proper OSS since August either, sorry.

From my end, the patch looks simple/solid

greg0ire commented:
Other similarity between failing jobs: `sudo: required`

greg0ire commented:
Possible solution: https://stackoverflow.com/questions/36133265/unable-to-get-env-variable-value-in-a-before-install-script-with-travis#36161645

OskarStark commented:
damn, nothing is working 👎

greg0ire commented:
You could try to add `sudo: required` to another build to see if this it what causes the problem

OskarStark commented:
updated

greg0ire commented:
Looks like it is! That's progress 👍

OskarStark commented:
> Looks like it is! That's progress 👍

yeah, we are happy, that we have more failing builds than before 😆

alcaeus commented:
> yeah, we are happy, that we have more failing builds than before

Sometimes you need to break things before you can fix them 🤣

greg0ire commented:
We are happy that we know more about the failure and what to google :P

lcobucci commented:
@OskarStark a stupid question: have you tried to set that environment variable in the phpunit configuration? I think that would be ideal also to ensure that the tests will not fail randomly for contributors with different configuration, what do you think?

greg0ire commented:
How is this stupid exactly? :P

lcobucci commented:
@greg0ire because he probably already tried it?

OskarStark commented:
> @OskarStark a stupid question: have you tried to set that environment variable in the phpunit configuration?

I didn't tried yet, but will give it a shot

greg0ire commented:
Looks like it wasn't stupid *at all* 🎉

OskarStark commented:
I squashed my commits!

@Ocramius @lcobucci this should be ready to merge, please let me know if I should change something!

silasjoisten reviewed (approved):
Nice! @OskarStark

javiereguiluz reviewed (approved):
I haven't tested the code for real ... but the changes look OK to me. Thanks for working on this Oskar!

lcobucci commented on .scrutinizer.yml:
```diff
@@ -1,5 +1,7 @@
 build:
     environment:
+        variables:
+            COLUMNS: 120
```
We're not running the tests on scrutinizer so I don't think this is needed

OskarStark replied:
I will try, but before, Scrutinizer was failing

OskarStark replied:
Looks, like tests were executed by Scrutinizer:
https://scrutinizer-ci.com/g/doctrine/doctrine2/inspections/fe506200-ebd0-465d-b9a1-1f4b7515fd8a

lcobucci replied:
Okay, I'll fix scrutinizer config, but it's failing because it's using the default phpunit config (`phpunit.xml.dist`) that hasn't been updated.

lcobucci replied:
@OskarStark #6843 fixes the scrutinizer configuration

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/ClearCache/MetadataCommand.php:
```diff
@@ -106,6 +107,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
-        $output->writeln($message);
+        if (!$result) {
```
This does not follow our code standards (there must be a space before and after the negation operator)

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/MappingDescribeCommand.php:
```diff
@@ -77,16 +80,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
      * @param EntityManagerInterface $entityManager
      * @param OutputInterface        $output
```
This parameter should be updated

lcobucci reviewed (changes requested):
Nice job, I got some requests for you:

* there are some parts of the code that are not being covered by tests, can you please check that? https://scrutinizer-ci.com/g/doctrine/doctrine2/inspections/cf59ead0-6eb9-4a28-b114-91d5328ba6e0/code-structure/test-coverage might be helpful
* You're formatting some parts of the code in the same commit of your changes, that's a bit confusing to review (specially via mobile :joy:), can you please revert those changes or split them into a different commit?
* there're some additions that don't follow our code standards (like the spacing before and after the negation operator), can you please check that? 
* I think you forgot to define the constant in  `phpunit.xml.dist`

OskarStark commented:
> there are some parts of the code that are not being covered by tests, can you please check that? https://scrutinizer-ci.com/g/doctrine/doctrine2/inspections/cf59ead0-6eb9-4a28-b114-91d5328ba6e0/code-structure/test-coverage might be helpful

I will have a look

> You're formatting some parts of the code in the same commit of your changes, that's a bit confusing to review (specially via mobile 😂), can you please revert those changes or split them into a different commit?

damn, i did before, but squashed all commits, thought this was already done 😞 
Shall I rework the PR and split everything in different commits?

> there're some additions that don't follow our code standards (like the spacing before and after the negation operator), can you please check that?

ofc, I will check, do you have a tool for that like StyleCi or a PHP-CS-Fixer config?

> I think you forgot to define the constant in phpunit.xml.dist

done

lcobucci commented:
> Shall I rework the PR and split everything in different commits?

I think it's fine to have 2 commits one that formats the old code and one that introduces `SymfonyStyle`, it should be simple to do that with reverting (soft) your commit and using `git add -p` but if I can help you on that later. 

> ofc, I will check, do you have a tool for that like StyleCi or a PHP-CS-Fixer config?

Yes, we use `phpcs` but we haven't configured it for the ORM yet 😭

OskarStark commented:
> I think it's fine to have 2 commits one that formats the old code and one that introduces SymfonyStyle, it should be simple to do that with reverting (soft) your commit and using git add -p but if I can help you on that later.

don't worry, I will do it, but how to proceed with Scrutinizer? set `COLUMNS` or not?

> Yes, we use phpcs but we haven't configured it for the ORM yet 😭

ok, i checked for the space before and after `!`

lcobucci commented:
> don't worry, I will do it, but how to proceed with Scrutinizer?

@OskarStark I've merged the scrutinizer config to not run the tests there, please rebase your branch to sync with `master`

> set COLUMNS or not?

Don't change anything on scrutinizer config =)

OskarStark committed:
Use addOption()/addArgument() to configure commands

In order to simplify and standardise the definition of the commands.

OskarStark committed:
Use SymfonyStyle for command output

OskarStark commented:
> @OskarStark I've merged the scrutinizer config to not run the tests there, please rebase your branch to sync with master

done

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/ClearCache/ResultCommand.php:
```diff
@@ -105,6 +100,12 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
-        $output->writeln($message);
+        if (!$result) {
```
You forgot to fix CS here

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheCollectionRegionCommandTest.php:
```diff
@@ -53,7 +53,10 @@ public function testClearAllRegion()
         );
 
-        $this->assertEquals('Clearing all second-level cache collection regions' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheCollectionRegionCommandTest.php:
```diff
@@ -68,7 +71,10 @@ public function testClearByOwnerEntityClassName()
         );
 
-        $this->assertEquals('Clearing second-level cache for collection "Doctrine\Tests\Models\Cache\State#cities"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheCollectionRegionCommandTest.php:
```diff
@@ -84,7 +90,14 @@ public function testClearCacheEntryName()
         );
 
-        $this->assertEquals('Clearing second-level cache entry for collection "Doctrine\Tests\Models\Cache\State#cities" owner entity identified by "1"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheCollectionRegionCommandTest.php:
```diff
@@ -84,7 +90,14 @@ public function testClearCacheEntryName()
+            ' // Clearing second-level cache entry for collection "Doctrine\Tests\Models\Cache\State#cities" owner',
+            $tester->getDisplay()
+        );
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheCollectionRegionCommandTest.php:
```diff
@@ -100,6 +113,9 @@ public function testFlushRegionName()
         );
 
-        $this->assertEquals('Flushing cache provider configured for "Doctrine\Tests\Models\Cache\State#cities"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheEntityRegionCommandTest.php:
```diff
@@ -53,7 +53,10 @@ public function testClearAllRegion()
         );
 
-        $this->assertEquals('Clearing all second-level cache entity regions' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheEntityRegionCommandTest.php:
```diff
@@ -67,7 +70,10 @@ public function testClearByEntityClassName()
         );
 
-        $this->assertEquals('Clearing second-level cache for entity "Doctrine\Tests\Models\Cache\Country"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheEntityRegionCommandTest.php:
```diff
@@ -82,7 +88,14 @@ public function testClearCacheEntryName()
         );
 
-        $this->assertEquals('Clearing second-level cache entry for entity "Doctrine\Tests\Models\Cache\Country" identified by "1"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheEntityRegionCommandTest.php:
```diff
@@ -82,7 +88,14 @@ public function testClearCacheEntryName()
+            ' // Clearing second-level cache entry for entity "Doctrine\Tests\Models\Cache\Country" identified by',
+            $tester->getDisplay()
+        );
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheEntityRegionCommandTest.php:
```diff
@@ -97,6 +110,9 @@ public function testFlushRegionName()
         );
 
-        $this->assertEquals('Flushing cache provider configured for entity named "Doctrine\Tests\Models\Cache\Country"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheQueryRegionCommandTest.php:
```diff
@@ -52,7 +52,10 @@ public function testClearAllRegion()
         );
 
-        $this->assertEquals('Clearing all second-level cache query regions' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheQueryRegionCommandTest.php:
```diff
@@ -66,7 +69,10 @@ public function testClearDefaultRegionName()
         );
 
-        $this->assertEquals('Clearing second-level cache query region named "query_cache_region"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheQueryRegionCommandTest.php:
```diff
@@ -80,7 +86,10 @@ public function testClearByRegionName()
         );
 
-        $this->assertEquals('Clearing second-level cache query region named "my_region"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/ClearCacheQueryRegionCommandTest.php:
```diff
@@ -95,6 +104,9 @@ public function testFlushRegionName()
         );
 
-        $this->assertEquals('Flushing cache provider configured for second-level cache query region named "my_region"' . PHP_EOL, $tester->getDisplay());
+        $this->assertContains(
```
Please use `self::assert*` instead

lcobucci commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/RunDqlCommandTest.php:
```diff
@@ -91,6 +91,6 @@ public function testWillShowQuery()
         );
 
-        $this->assertStringMatchesFormat('%Astring%sSELECT %a', $this->tester->getDisplay());
+        $this->assertStringMatchesFormat('SELECT %a', trim($this->tester->getDisplay()));
```
Please use `self::assert*` instead

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/InfoCommand.php:
```diff
@@ -54,32 +55,39 @@ protected function configure()
         if (!$entityClassNames) {
-            throw new \Exception(
-                'You do not have any mapped Doctrine ORM entities according to the current configuration. '.
+            $ui->caution([
```
There's no test covering this, could you please add one?

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/InfoCommand.php:
```diff
@@ -54,32 +55,39 @@ protected function configure()
-                $output->writeln("<error>[FAIL]</error> ".$entityClassName);
-                $output->writeln(sprintf("<comment>%s</comment>", $e->getMessage()));
-                $output->writeln('');
+                $ui->text([
```
There's no test covering this, could you please add one?

lcobucci commented on lib/Doctrine/ORM/Tools/Console/Command/GenerateRepositoriesCommand.php:
```diff
@@ -112,12 +106,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
             }
         } else {
-            $output->writeln('No Metadata Classes to process.');
+            $ui->success('No Metadata Classes to process.');
```
There's no test covering this, could you please add one?

lcobucci reviewed (changes requested):
@OskarStark great job with the commit organisation (and thanks for fixing gramar mistakes as well)!

I got some **final** nitpickings. For things related coding standard stuff I've just merged https://github.com/doctrine/doctrine2/pull/6844 in, so that might help you check for possible issues (along with the [tool that @morozov created](https://github.com/morozov/diff-sniffer-pull-request)).

lcobucci commented:
@OskarStark thanks a lot for not giving up on this and sorry about the huge delays. We'll probably get this 🚢  by the end of the day!

OskarStark commented:
> @OskarStark thanks a lot for not giving up on this and sorry about the huge delays. We'll probably get this 🚢 by the end of the day!

sounds good 😄 

- [ ] I added 2 tests for `InfoCommand` in a separate Command
- [ ] I switched from `$this->assert*` to `self::assert*` in a separate commit
- [ ] I removed all `setDefinition()` calls and replaced them by `addOption()` and `addArgument()` in a separate commit
- [ ] I will now write a test for `GenerateRepositoriesCommand`

In the meantime you could give me another review if you find some time.

Cheers 🍾

OskarStark commented on tests/Doctrine/Tests/ORM/Tools/Console/Command/InfoCommandTest.php:
```diff
@@ -41,15 +45,112 @@ protected function setUp()
+        self::assertContains(City::class, $this->tester->getDisplay());
+    }
+
+    public function testEmptyEntityClassNames()
```
I tried to overwrite the `HelperSet`with my mocked `$em` like this, but it doesn't work:
```php
// overwrite HelperSet to mock getAllClassNames()
         $this->application->setHelperSet(
             new HelperSet(['em' => new EntityManagerHelper($em)])
         );
```

because of this I need to redo the whole `setUp` logic in my tests

lcobucci replied:
okay, let's give it a shot but if it gets too complicated it means that we have a design issue and wouldn't be really correct to fix it in the PR (trade-offs)

OskarStark replied:
I am done with the test, but will force push once you finished your review.

lcobucci replied:
You can push it, I didn't start the review yet

OskarStark replied:
done

OskarStark commented:
I will keep the separate commits, let me know if I should squash some of them together! 👍

lcobucci committed:
Remove constants existence validation

These constants are available since PHP 5.4 and since we're requiring
PHP 7.1 there's no reason to require them.

I've also simplified the `array_map()` call since it's useless to
define a closure that simply calls a function.

lcobucci committed:
Format commands' tests

To simplify and organise the code (also replacing `$this->assert*`
with `self::assert*`).

lcobucci reviewed (changes requested):
I've found some small issues, which I've already fixed 😄 (just to keep track of why I've introduced new changes here)

lcobucci commented:
@OskarStark 🚢!

Thanks a lot for your contribution (and patience)!!

OskarStark commented:
Yeah, finally 😃 🎉 🍾 

Thank you for your detailed feedback!

hiddewie commented:
It's a shame I never got a reaction for PR #5651, posted over two years ago. I saw this PR via Twitter only today.

Majkl578 commented on lib/Doctrine/ORM/Tools/Console/Command/GenerateProxiesCommand.php:
```diff
@@ -98,20 +89,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
+
+        // Outputting information message
+        $ui->newLine();
+        $ui->text(sprintf('Proxy classes generated to "<info>%s</INFO>"', $destPath));
```
Any reason why this one and the above are left out?

OskarStark replied:
I don't get what you mean

Majkl578 replied:
I mean those `<info>` tags, I thought whole point of this PR is to eliminate/replace them with SymfonyStyle.

OskarStark replied:
we keep them, because the dest path should be green, and the rest of the text white
